content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java

```diff
@@ -10384,5 +10412,362 @@ public void doViewTrash(RunData data)
+				{
+					zipDownloadItems.add(item);
+				}
+			}
+			catch(PermissionException e)
+			{
+				logger.warn("PermissionException: "+e,e);
+				System.out.println("PERMISSION EXCEPTION. "+e);
+			}
+			catch (IdUnusedException e)
+			{
+				logger.warn("IdUnusedException: "+e,e);
+				System.out.println("IDUNUSEDEXCEPTION. "+e);
+			}
+			catch (TypeException e)
+			{
+				logger.warn("TypeException: "+e,e);
+				System.out.println("TYPEEXCEPTION. "+e);
+			}
+			catch (Exception e)
+			{
+				System.out.println("Exception: "+e);
+			}
+		}
+
+		state.setAttribute (STATE_ZIPDOWNLOAD_SET, zipDownloadItems);
+	}
+
+	/**
+	 * @param portlet
+	 * @param context
+	 * @param data
+	 * @param state
+	 * @return
+	 */
+	public String buildZipDownloadFinishContext(VelocityPortlet portlet, Context context, RunData data, SessionState state)
+	{
+		context.put("tlang",trb);
+		context.put ("collectionId", state.getAttribute (STATE_COLLECTION_ID) );
+
+		List zipDownloadItems = (List) state.getAttribute(STATE_ZIPDOWNLOAD_SET);
+		context.put ("zipDownloadItems", zipDownloadItems);
+
+		return TEMPLATE_ZIPDOWNLOAD_FINISH;
+	}
+
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException
+	{
+		String action = request.getParameter("eventSubmit_doFinalizeZipDownload");
+
+		if ((action==null)||(action.isEmpty()))
+		{
+			super.doPost(request,response);
+			return;
+		}
+
+		checkRunData(request);
+		JetspeedRunData rundata = (JetspeedRunData) request.getAttribute(ATTR_RUNDATA);
+		SessionState state = rundata.getPortletSessionState (rundata.getJs_peid ());
+		List<ListItem> zipDownloadItems = (List<ListItem>) state.getAttribute(STATE_ZIPDOWNLOAD_SET);
+
+		String collectionId = (String) request.getParameter("collectionId");
+		ZipOutputStream zipOut = null;
+		try
+		{
+			ContentCollection collection = ContentHostingService.getCollection(collectionId);
+			ResourceProperties props = collection.getProperties();
+			String rootFolderName = escapeInvalidCharsEntry(props.getPropertyFormatted(props.getNamePropDisplayName()));
+
+			response.setContentType("application/zip;charset=UTF-8");
+			response.setHeader("Content-Disposition", "attachment; filename = site_download.zip");
+			zipOut = new ZipOutputStream(response.getOutputStream());
+
+			Iterator it = zipDownloadItems.iterator();
+			while(it.hasNext())
+			{
+				ListItem myElement = (ListItem) it.next();
+				String resourceId = myElement.getId();
+				boolean get = ContentHostingService.allowGetResource(resourceId);
+				if (get) compressResource(zipOut, collectionId, rootFolderName, resourceId);
+			}
+		}
+		catch (PermissionException pe)
+		{
+			try
+			{
+				response.sendError(response.SC_FORBIDDEN);
+			}
+			catch (IOException e)
+			{
+				e.printStackTrace();
+			}
+		}
+		catch (Throwable ignore)
+		{
+			try
+			{
+				response.sendError(response.SC_NO_CONTENT);
+			}
+			catch (IOException e)
+			{
+				e.printStackTrace();
```

Comment:
Production code should never use `e.printStackTrace()` as it can end up going to a different location to the logs and doesn't include all the useful features the loggers give (timestamps, extra context, etc). Could you switch this to use the logging framework please.